{"created": 1756178955.468923, "duration": 11.733449935913086, "exitcode": 1, "root": "/home/marku/Documents/programming/LocalProgramming", "environment": {}, "summary": {"passed": 63, "failed": 9, "total": 72, "collected": 72}, "collectors": [{"nodeid": "", "outcome": "passed", "result": [{"nodeid": "tests/unit", "type": "Dir"}]}, {"nodeid": "tests/unit/test_base_provider.py::TestModelInfo", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestModelInfo::test_model_info_creation", "type": "Function", "lineno": 20}, {"nodeid": "tests/unit/test_base_provider.py::TestModelInfo::test_model_info_optional_fields", "type": "Function", "lineno": 36}]}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionRequest", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestCompletionRequest::test_completion_request_basic", "type": "Function", "lineno": 51}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionRequest::test_completion_request_full", "type": "Function", "lineno": 66}]}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionResponse", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestCompletionResponse::test_completion_response_basic", "type": "Function", "lineno": 89}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionResponse::test_completion_response_full", "type": "Function", "lineno": 105}]}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_provider_initialization", "type": "Function", "lineno": 171}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_provider_name_extraction", "type": "Function", "lineno": 178}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_provider_initialize", "type": "Coroutine", "lineno": 182}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_list_models", "type": "Coroutine", "lineno": 189}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_complete", "type": "Coroutine", "lineno": 197}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_stream_complete", "type": "Coroutine", "lineno": 210}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_health_check", "type": "Coroutine", "lineno": 225}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_count_tokens_default", "type": "Coroutine", "lineno": 239}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_estimate_cost_default", "type": "Coroutine", "lineno": 249}]}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration::test_local_provider_config", "type": "Function", "lineno": 259}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration::test_remote_provider_config", "type": "Function", "lineno": 277}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration::test_missing_config_handling", "type": "Function", "lineno": 296}]}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling::test_initialization_failure", "type": "Coroutine", "lineno": 315}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling::test_completion_failure", "type": "Coroutine", "lineno": 327}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling::test_streaming_failure", "type": "Coroutine", "lineno": 343}]}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderIntegration", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestProviderIntegration::test_full_workflow", "type": "Coroutine", "lineno": 370}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderIntegration::test_concurrent_requests", "type": "Coroutine", "lineno": 398}]}, {"nodeid": "tests/unit/test_base_provider.py", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py::TestModelInfo", "type": "Class"}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionRequest", "type": "Class"}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionResponse", "type": "Class"}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider", "type": "Class"}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration", "type": "Class"}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling", "type": "Class"}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderIntegration", "type": "Class"}]}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_base_provider_is_abstract", "type": "Function", "lineno": 49}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_provider_name_extraction", "type": "Function", "lineno": 54}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_default_token_counting", "type": "Function", "lineno": 66}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_default_cost_estimation", "type": "Function", "lineno": 82}]}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_provider_initialization", "type": "Function", "lineno": 108}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_server_connection_success", "type": "Coroutine", "lineno": 115}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_server_connection_failure", "type": "Coroutine", "lineno": 132}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_server_http_error", "type": "Coroutine", "lineno": 142}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_list_models_success", "type": "Coroutine", "lineno": 154}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_list_models_empty_response", "type": "Coroutine", "lineno": 190}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_list_models_error", "type": "Coroutine", "lineno": 201}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_non_streaming_completion", "type": "Coroutine", "lineno": 210}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_streaming_completion", "type": "Coroutine", "lineno": 241}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_health_check_healthy", "type": "Coroutine", "lineno": 270}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_health_check_unhealthy", "type": "Coroutine", "lineno": 286}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_completion_error_handling", "type": "Coroutine", "lineno": 299}]}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_provider_manager_initialization", "type": "Function", "lineno": 324}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_provider_initialization", "type": "Coroutine", "lineno": 330}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_provider_initialization_with_failures", "type": "Coroutine", "lineno": 339}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_fallback_completion_success", "type": "Coroutine", "lineno": 348}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_fallback_completion_with_primary_failure", "type": "Coroutine", "lineno": 364}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_fallback_all_providers_fail", "type": "Coroutine", "lineno": 385}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_get_all_models", "type": "Coroutine", "lineno": 400}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_health_check_all", "type": "Coroutine", "lineno": 423}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_get_provider", "type": "Function", "lineno": 442}]}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_model_info_creation", "type": "Function", "lineno": 458}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_model_info_defaults", "type": "Function", "lineno": 474}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_completion_request_creation", "type": "Function", "lineno": 485}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_completion_request_defaults", "type": "Function", "lineno": 502}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_completion_response_creation", "type": "Function", "lineno": 515}]}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderIntegration", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderIntegration::test_provider_roundtrip", "type": "Coroutine", "lineno": 537}]}, {"nodeid": "tests/unit/test_provider_interfaces.py", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider", "type": "Class"}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider", "type": "Class"}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager", "type": "Class"}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels", "type": "Class"}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderIntegration", "type": "Class"}]}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_manager_initialization", "type": "Function", "lineno": 27}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_initialize_providers_success", "type": "Coroutine", "lineno": 34}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_initialize_providers_partial_failure", "type": "Coroutine", "lineno": 57}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_initialize_providers_missing_config", "type": "Coroutine", "lineno": 71}]}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_success", "type": "Coroutine", "lineno": 110}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_chain", "type": "Coroutine", "lineno": 127}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_all_fail", "type": "Coroutine", "lineno": 153}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_unhealthy_providers", "type": "Coroutine", "lineno": 173}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_get_provider", "type": "Function", "lineno": 190}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_get_all_models", "type": "Coroutine", "lineno": 198}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_get_all_models_with_failures", "type": "Coroutine", "lineno": 211}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_health_check_all", "type": "Coroutine", "lineno": 227}]}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic::test_fallback_order_preferred_provider", "type": "Coroutine", "lineno": 249}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic::test_fallback_with_mixed_health_states", "type": "Coroutine", "lineno": 294}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic::test_provider_error_handling", "type": "Coroutine", "lineno": 323}]}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance::test_concurrent_requests", "type": "Coroutine", "lineno": 364}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance::test_provider_selection_performance", "type": "Coroutine", "lineno": 394}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance::test_memory_usage_with_many_providers", "type": "Coroutine", "lineno": 425}]}, {"nodeid": "tests/unit/test_provider_manager.py", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager", "type": "Class"}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks", "type": "Class"}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic", "type": "Class"}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance", "type": "Class"}]}, {"nodeid": "tests/unit", "outcome": "passed", "result": [{"nodeid": "tests/unit/test_base_provider.py", "type": "Module"}, {"nodeid": "tests/unit/test_provider_interfaces.py", "type": "Module"}, {"nodeid": "tests/unit/test_provider_manager.py", "type": "Module"}]}], "tests": [{"nodeid": "tests/unit/test_base_provider.py::TestModelInfo::test_model_info_creation", "lineno": 20, "outcome": "passed", "keywords": ["test_model_info_creation", "TestModelInfo", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.000690341999870725, "outcome": "passed"}, "call": {"duration": 0.00029661999724339694, "outcome": "passed"}, "teardown": {"duration": 0.00023201698786579072, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestModelInfo::test_model_info_optional_fields", "lineno": 36, "outcome": "passed", "keywords": ["test_model_info_optional_fields", "TestModelInfo", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00018870600615628064, "outcome": "passed"}, "call": {"duration": 0.00019142100063618273, "outcome": "passed"}, "teardown": {"duration": 0.00013295099779497832, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionRequest::test_completion_request_basic", "lineno": 51, "outcome": "passed", "keywords": ["test_completion_request_basic", "TestCompletionRequest", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00017742499767336994, "outcome": "passed"}, "call": {"duration": 0.00019175100896973163, "outcome": "passed"}, "teardown": {"duration": 0.00012284099648240954, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionRequest::test_completion_request_full", "lineno": 66, "outcome": "passed", "keywords": ["test_completion_request_full", "TestCompletionRequest", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00015948100190144032, "outcome": "passed"}, "call": {"duration": 0.00018313499458599836, "outcome": "passed"}, "teardown": {"duration": 0.00012547700316645205, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionResponse::test_completion_response_basic", "lineno": 89, "outcome": "passed", "keywords": ["test_completion_response_basic", "TestCompletionResponse", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00015379999240394682, "outcome": "passed"}, "call": {"duration": 0.00018556999566499144, "outcome": "passed"}, "teardown": {"duration": 0.0001232229988090694, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestCompletionResponse::test_completion_response_full", "lineno": 105, "outcome": "passed", "keywords": ["test_completion_response_full", "TestCompletionResponse", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00014903099508956075, "outcome": "passed"}, "call": {"duration": 0.00018023999291472137, "outcome": "passed"}, "teardown": {"duration": 0.00012577700545080006, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_provider_initialization", "lineno": 171, "outcome": "passed", "keywords": ["test_provider_initialization", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0003116979933110997, "outcome": "passed"}, "call": {"duration": 0.0001777159923221916, "outcome": "passed"}, "teardown": {"duration": 0.00015504298789892346, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_provider_name_extraction", "lineno": 178, "outcome": "passed", "keywords": ["test_provider_name_extraction", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0002561229921411723, "outcome": "passed"}, "call": {"duration": 0.00017245499475393444, "outcome": "passed"}, "teardown": {"duration": 0.00015186700329650193, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_provider_initialize", "lineno": 182, "outcome": "passed", "keywords": ["test_provider_initialize", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0011855159973492846, "outcome": "passed"}, "call": {"duration": 0.0005067260062787682, "outcome": "passed"}, "teardown": {"duration": 0.0019006759976036847, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_list_models", "lineno": 189, "outcome": "passed", "keywords": ["test_list_models", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006746720027877018, "outcome": "passed"}, "call": {"duration": 0.0003707589930854738, "outcome": "passed"}, "teardown": {"duration": 0.0003604399971663952, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_complete", "lineno": 197, "outcome": "passed", "keywords": ["test_complete", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.000498961002449505, "outcome": "passed"}, "call": {"duration": 0.00036142200406175107, "outcome": "passed"}, "teardown": {"duration": 0.0003183700027875602, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_stream_complete", "lineno": 210, "outcome": "passed", "keywords": ["test_stream_complete", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005375439941417426, "outcome": "passed"}, "call": {"duration": 0.00031180800579022616, "outcome": "passed"}, "teardown": {"duration": 0.00032995200308505446, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_health_check", "lineno": 225, "outcome": "passed", "keywords": ["test_health_check", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0004692149959737435, "outcome": "passed"}, "call": {"duration": 0.00028228299925103784, "outcome": "passed"}, "teardown": {"duration": 0.00030053600494284183, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_count_tokens_default", "lineno": 239, "outcome": "passed", "keywords": ["test_count_tokens_default", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00041831900307442993, "outcome": "passed"}, "call": {"duration": 0.000300157000310719, "outcome": "passed"}, "teardown": {"duration": 0.00029114999051671475, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestBaseProvider::test_estimate_cost_default", "lineno": 249, "outcome": "passed", "keywords": ["test_estimate_cost_default", "asyncio", "pytestmark", "TestBaseProvider", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0004228770121699199, "outcome": "passed"}, "call": {"duration": 0.0002644380001584068, "outcome": "passed"}, "teardown": {"duration": 0.00030514599347952753, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration::test_local_provider_config", "lineno": 259, "outcome": "passed", "keywords": ["test_local_provider_config", "TestProviderConfiguration", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00016454000433441252, "outcome": "passed"}, "call": {"duration": 0.00023656600387766957, "outcome": "passed"}, "teardown": {"duration": 0.0001370289974147454, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration::test_remote_provider_config", "lineno": 277, "outcome": "passed", "keywords": ["test_remote_provider_config", "TestProviderConfiguration", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0001583989942446351, "outcome": "passed"}, "call": {"duration": 0.00022084699594415724, "outcome": "passed"}, "teardown": {"duration": 0.0001368990051560104, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderConfiguration::test_missing_config_handling", "lineno": 296, "outcome": "passed", "keywords": ["test_missing_config_handling", "TestProviderConfiguration", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.000156765992869623, "outcome": "passed"}, "call": {"duration": 0.00018131198885384947, "outcome": "passed"}, "teardown": {"duration": 0.00014103599824011326, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling::test_initialization_failure", "lineno": 315, "outcome": "passed", "keywords": ["test_initialization_failure", "asyncio", "pytestmark", "TestProviderErrorHandling", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005638330039801076, "outcome": "passed"}, "call": {"duration": 0.0016409649979323149, "outcome": "passed"}, "teardown": {"duration": 0.00047663900477346033, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling::test_completion_failure", "lineno": 327, "outcome": "passed", "keywords": ["test_completion_failure", "asyncio", "pytestmark", "TestProviderErrorHandling", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005624609912047163, "outcome": "passed"}, "call": {"duration": 0.0010134009935427457, "outcome": "passed"}, "teardown": {"duration": 0.00032530400494579226, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderErrorHandling::test_streaming_failure", "lineno": 343, "outcome": "passed", "keywords": ["test_streaming_failure", "asyncio", "pytestmark", "TestProviderErrorHandling", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00045045999286230654, "outcome": "passed"}, "call": {"duration": 0.0009595599985914305, "outcome": "passed"}, "teardown": {"duration": 0.0003199740021955222, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderIntegration::test_full_workflow", "lineno": 370, "outcome": "failed", "keywords": ["test_full_workflow", "asyncio", "pytestmark", "TestProviderIntegration", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0003726029972312972, "outcome": "passed"}, "call": {"duration": 0.00033291800355073065, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_base_provider.py", "lineno": 385, "message": "AttributeError: 'TestProviderIntegration' object has no attribute 'provider'"}, "traceback": [{"path": "tests/unit/test_base_provider.py", "lineno": 385, "message": "in test_full_workflow"}], "longrepr": "tests/unit/test_base_provider.py:385: in test_full_workflow\n    models = await self.provider.list_models()\n                   ^^^^^^^^^^^^^\nE   AttributeError: 'TestProviderIntegration' object has no attribute 'provider'"}, "teardown": {"duration": 0.00055200100177899, "outcome": "passed"}}, {"nodeid": "tests/unit/test_base_provider.py::TestProviderIntegration::test_concurrent_requests", "lineno": 398, "outcome": "passed", "keywords": ["test_concurrent_requests", "asyncio", "pytestmark", "TestProviderIntegration", "test_base_provider.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005084190052002668, "outcome": "passed"}, "call": {"duration": 0.0004937320045428351, "outcome": "passed"}, "teardown": {"duration": 0.00032244798785541207, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_base_provider_is_abstract", "lineno": 49, "outcome": "passed", "keywords": ["test_base_provider_is_abstract", "TestBaseProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0001803799968911335, "outcome": "passed"}, "call": {"duration": 0.00023725698702037334, "outcome": "passed"}, "teardown": {"duration": 0.0001355459971819073, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_provider_name_extraction", "lineno": 54, "outcome": "passed", "keywords": ["test_provider_name_extraction", "TestBaseProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00016538200725335628, "outcome": "passed"}, "call": {"duration": 0.00024152600963134319, "outcome": "passed"}, "teardown": {"duration": 0.00013108699931763113, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_default_token_counting", "lineno": 66, "outcome": "failed", "keywords": ["test_default_token_counting", "TestBaseProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00016061299538705498, "outcome": "passed"}, "call": {"duration": 0.0005830500012962148, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_provider_interfaces.py", "lineno": 80, "message": "AssertionError: assert False\n +  where False = isinstance(<coroutine object BaseProvider.count_tokens at 0x75dc737559a0>, (<class 'int'>, <class 'float'>))"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 80, "message": "in test_default_token_counting"}], "longrepr": "tests/unit/test_provider_interfaces.py:80: in test_default_token_counting\n    assert isinstance(token_count, (int, float))\nE   AssertionError: assert False\nE    +  where False = isinstance(<coroutine object BaseProvider.count_tokens at 0x75dc737559a0>, (<class 'int'>, <class 'float'>))"}, "teardown": {"duration": 0.00025832800019998103, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestBaseProvider::test_default_cost_estimation", "lineno": 82, "outcome": "failed", "keywords": ["test_default_cost_estimation", "TestBaseProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0001981040113605559, "outcome": "passed"}, "call": {"duration": 0.00046116999874357134, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_provider_interfaces.py", "lineno": 96, "message": "assert <coroutine object BaseProvider.estimate_cost at 0x75dc7375f5e0> == 0.0"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 96, "message": "in test_default_cost_estimation"}], "longrepr": "tests/unit/test_provider_interfaces.py:96: in test_default_cost_estimation\n    assert cost == 0.0\nE   assert <coroutine object BaseProvider.estimate_cost at 0x75dc7375f5e0> == 0.0"}, "teardown": {"duration": 0.00026220400468446314, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_provider_initialization", "lineno": 108, "outcome": "passed", "keywords": ["test_ollama_provider_initialization", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.000448606995632872, "outcome": "passed"}, "call": {"duration": 0.00026236499252263457, "outcome": "passed"}, "teardown": {"duration": 0.00017887799185700715, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_server_connection_success", "lineno": 115, "outcome": "passed", "keywords": ["test_ollama_server_connection_success", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006763650017092004, "outcome": "passed"}, "call": {"duration": 0.0043665359989972785, "outcome": "passed"}, "teardown": {"duration": 0.00043464898772072047, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_server_connection_failure", "lineno": 132, "outcome": "passed", "keywords": ["test_ollama_server_connection_failure", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0007618070085300133, "outcome": "passed"}, "call": {"duration": 0.0009952069958671927, "outcome": "passed"}, "teardown": {"duration": 0.0003645679971668869, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_ollama_server_http_error", "lineno": 142, "outcome": "passed", "keywords": ["test_ollama_server_http_error", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005445069982670248, "outcome": "passed"}, "call": {"duration": 0.0015820639964658767, "outcome": "passed"}, "teardown": {"duration": 0.00036359600198920816, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_list_models_success", "lineno": 154, "outcome": "passed", "keywords": ["test_list_models_success", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005816579941892996, "outcome": "passed"}, "call": {"duration": 0.0018545089988037944, "outcome": "passed"}, "teardown": {"duration": 0.0004235290107317269, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_list_models_empty_response", "lineno": 190, "outcome": "passed", "keywords": ["test_list_models_empty_response", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006188569968799129, "outcome": "passed"}, "call": {"duration": 0.0020256309944670647, "outcome": "passed"}, "teardown": {"duration": 0.00045170199882704765, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_list_models_error", "lineno": 201, "outcome": "passed", "keywords": ["test_list_models_error", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005815460026497021, "outcome": "passed"}, "call": {"duration": 0.0010470550041645765, "outcome": "passed", "stdout": "Error listing Ollama models: Network error\n"}, "teardown": {"duration": 0.0003952060069423169, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_non_streaming_completion", "lineno": 210, "outcome": "failed", "keywords": ["test_non_streaming_completion", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00086300699331332, "outcome": "passed"}, "call": {"duration": 0.002320767001947388, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/app/llm_providers/ollama_provider.py", "lineno": 68, "message": "KeyError: 'message'"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 232, "message": "in test_non_streaming_completion"}, {"path": "app/llm_providers/ollama_provider.py", "lineno": 68, "message": "in complete"}], "longrepr": "tests/unit/test_provider_interfaces.py:232: in test_non_streaming_completion\n    response = await ollama_provider.complete(sample_completion_request)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\napp/llm_providers/ollama_provider.py:68: in complete\n    content=data['message']['content'],\n            ^^^^^^^^^^^^^^^\nE   KeyError: 'message'"}, "teardown": {"duration": 0.0005993410013616085, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_streaming_completion", "lineno": 241, "outcome": "failed", "keywords": ["test_streaming_completion", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0008385620021726936, "outcome": "passed"}, "call": {"duration": 0.003583008990972303, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_provider_interfaces.py", "lineno": 269, "message": "AssertionError: assert [] == ['Hello', ' there', '!']\n  \n  Right contains 3 more items, first extra item: 'Hello'\n  \n  Full diff:\n  + []\n  - [\n  -     'Hello',...\n  \n  ...Full output truncated (3 lines hidden), use '-vv' to show"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 269, "message": "in test_streaming_completion"}], "longrepr": "tests/unit/test_provider_interfaces.py:269: in test_streaming_completion\n    assert chunks == [\"Hello\", \" there\", \"!\"]\nE   AssertionError: assert [] == ['Hello', ' there', '!']\nE     \nE     Right contains 3 more items, first extra item: 'Hello'\nE     \nE     Full diff:\nE     + []\nE     - [\nE     -     'Hello',...\nE     \nE     ...Full output truncated (3 lines hidden), use '-vv' to show"}, "teardown": {"duration": 0.0005788519920315593, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_health_check_healthy", "lineno": 270, "outcome": "failed", "keywords": ["test_health_check_healthy", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006691120070172474, "outcome": "passed"}, "call": {"duration": 0.0021782389958389103, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_provider_interfaces.py", "lineno": 284, "message": "AssertionError: assert 'latency_ms' in {'healthy': True, 'models': ['llama2:7b'], 'models_available': 1, 'provider': 'ollama'}"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 284, "message": "in test_health_check_healthy"}], "longrepr": "tests/unit/test_provider_interfaces.py:284: in test_health_check_healthy\n    assert \"latency_ms\" in health\nE   AssertionError: assert 'latency_ms' in {'healthy': True, 'models': ['llama2:7b'], 'models_available': 1, 'provider': 'ollama'}"}, "teardown": {"duration": 0.0008849590085446835, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_health_check_unhealthy", "lineno": 286, "outcome": "failed", "keywords": ["test_health_check_unhealthy", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0007285739993676543, "outcome": "passed"}, "call": {"duration": 0.0012326149881118909, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_provider_interfaces.py", "lineno": 297, "message": "AssertionError: assert 'error' in {'healthy': False, 'models': [], 'models_available': 0, 'provider': 'ollama'}"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 297, "message": "in test_health_check_unhealthy"}], "longrepr": "tests/unit/test_provider_interfaces.py:297: in test_health_check_unhealthy\n    assert \"error\" in health\nE   AssertionError: assert 'error' in {'healthy': False, 'models': [], 'models_available': 0, 'provider': 'ollama'}"}, "teardown": {"duration": 0.0006360999977914616, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestOllamaProvider::test_completion_error_handling", "lineno": 299, "outcome": "passed", "keywords": ["test_completion_error_handling", "asyncio", "pytestmark", "TestOllamaProvider", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0010862390045076609, "outcome": "passed"}, "call": {"duration": 0.0013179660018067807, "outcome": "passed"}, "teardown": {"duration": 0.0005025079881306738, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_provider_manager_initialization", "lineno": 324, "outcome": "passed", "keywords": ["test_provider_manager_initialization", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0003853459929814562, "outcome": "passed"}, "call": {"duration": 0.00023248900834005326, "outcome": "passed"}, "teardown": {"duration": 0.0001753109972923994, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_provider_initialization", "lineno": 330, "outcome": "passed", "keywords": ["test_provider_initialization", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005968149926047772, "outcome": "passed"}, "call": {"duration": 1.4044736579962773, "outcome": "passed", "stdout": "Failed to initialize OpenAI: Error code: 401 - {'error': {'message': 'Incorrect API key provided: test-key. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}\nProvider openai is not available\n"}, "teardown": {"duration": 0.001762895000865683, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_provider_initialization_with_failures", "lineno": 339, "outcome": "passed", "keywords": ["test_provider_initialization_with_failures", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0009719430090626702, "outcome": "passed"}, "call": {"duration": 0.7780922579986509, "outcome": "passed", "stdout": "Failed to initialize OpenAI: Error code: 401 - {'error': {'message': 'Incorrect API key provided: test-key. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}\nProvider ollama is not available\nProvider openai is not available\n"}, "teardown": {"duration": 0.0008549519989173859, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_fallback_completion_success", "lineno": 348, "outcome": "passed", "keywords": ["test_fallback_completion_success", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0008321699860971421, "outcome": "passed"}, "call": {"duration": 0.0010638170060701668, "outcome": "passed"}, "teardown": {"duration": 0.00043995000305585563, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_fallback_completion_with_primary_failure", "lineno": 364, "outcome": "passed", "keywords": ["test_fallback_completion_with_primary_failure", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0007330419903155416, "outcome": "passed"}, "call": {"duration": 0.0018000860000029206, "outcome": "passed"}, "teardown": {"duration": 0.0005224860069574788, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_fallback_all_providers_fail", "lineno": 385, "outcome": "passed", "keywords": ["test_fallback_all_providers_fail", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0007082149968482554, "outcome": "passed"}, "call": {"duration": 0.002049376009381376, "outcome": "passed"}, "teardown": {"duration": 0.0005215329874772578, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_get_all_models", "lineno": 400, "outcome": "passed", "keywords": ["test_get_all_models", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005802139930892736, "outcome": "passed"}, "call": {"duration": 0.0011761480127461255, "outcome": "passed"}, "teardown": {"duration": 0.00047256100515369326, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_health_check_all", "lineno": 423, "outcome": "passed", "keywords": ["test_health_check_all", "asyncio", "pytestmark", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006321520049823448, "outcome": "passed"}, "call": {"duration": 0.0011852750030811876, "outcome": "passed"}, "teardown": {"duration": 0.00037960600457154214, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderManager::test_get_provider", "lineno": 442, "outcome": "passed", "keywords": ["test_get_provider", "TestProviderManager", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00034911900002043694, "outcome": "passed"}, "call": {"duration": 0.0002530880010453984, "outcome": "passed"}, "teardown": {"duration": 0.00028632101020775735, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_model_info_creation", "lineno": 458, "outcome": "passed", "keywords": ["test_model_info_creation", "TestDataModels", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0003368150064488873, "outcome": "passed"}, "call": {"duration": 0.00034887800575233996, "outcome": "passed"}, "teardown": {"duration": 0.0002521150017855689, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_model_info_defaults", "lineno": 474, "outcome": "passed", "keywords": ["test_model_info_defaults", "TestDataModels", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0003446299961069599, "outcome": "passed"}, "call": {"duration": 0.0004174080095253885, "outcome": "passed"}, "teardown": {"duration": 0.0002739159972406924, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_completion_request_creation", "lineno": 485, "outcome": "passed", "keywords": ["test_completion_request_creation", "TestDataModels", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00028657099755946547, "outcome": "passed"}, "call": {"duration": 0.0003234410105505958, "outcome": "passed"}, "teardown": {"duration": 0.0002477980015100911, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_completion_request_defaults", "lineno": 502, "outcome": "passed", "keywords": ["test_completion_request_defaults", "TestDataModels", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00029217099654488266, "outcome": "passed"}, "call": {"duration": 0.0003616419999161735, "outcome": "passed"}, "teardown": {"duration": 0.00024402000417467207, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestDataModels::test_completion_response_creation", "lineno": 515, "outcome": "passed", "keywords": ["test_completion_response_creation", "TestDataModels", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00035411799035500735, "outcome": "passed"}, "call": {"duration": 0.00038872400182299316, "outcome": "passed"}, "teardown": {"duration": 0.00026703299954533577, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_interfaces.py::TestProviderIntegration::test_provider_roundtrip", "lineno": 537, "outcome": "failed", "keywords": ["test_provider_roundtrip", "asyncio", "pytestmark", "TestProviderIntegration", "integration", "test_provider_interfaces.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0007663459982722998, "outcome": "passed"}, "call": {"duration": 0.004538370005320758, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/app/llm_providers/ollama_provider.py", "lineno": 68, "message": "KeyError: 'message'"}, "traceback": [{"path": "tests/unit/test_provider_interfaces.py", "lineno": 573, "message": "in test_provider_roundtrip"}, {"path": "app/llm_providers/ollama_provider.py", "lineno": 68, "message": "in complete"}], "longrepr": "tests/unit/test_provider_interfaces.py:573: in test_provider_roundtrip\n    response = await provider.complete(request)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\napp/llm_providers/ollama_provider.py:68: in complete\n    content=data['message']['content'],\n            ^^^^^^^^^^^^^^^\nE   KeyError: 'message'"}, "teardown": {"duration": 0.0006242579984245822, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_manager_initialization", "lineno": 27, "outcome": "passed", "keywords": ["test_manager_initialization", "TestProviderManager", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0003646580007625744, "outcome": "passed"}, "call": {"duration": 0.00020255199342500418, "outcome": "passed"}, "teardown": {"duration": 0.0001515460025984794, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_initialize_providers_success", "lineno": 34, "outcome": "passed", "keywords": ["test_initialize_providers_success", "asyncio", "pytestmark", "TestProviderManager", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005394170002546161, "outcome": "passed"}, "call": {"duration": 0.0018437379912938923, "outcome": "passed"}, "teardown": {"duration": 0.00040606599941384047, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_initialize_providers_partial_failure", "lineno": 57, "outcome": "passed", "keywords": ["test_initialize_providers_partial_failure", "asyncio", "pytestmark", "TestProviderManager", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005133980012033135, "outcome": "passed"}, "call": {"duration": 0.9215464789886028, "outcome": "passed", "stdout": "Failed to initialize OpenAI: Error code: 401 - {'error': {'message': 'Incorrect API key provided: sk-test123. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}\nFailed to initialize ollama: Connection failed\nProvider openai is not available\n"}, "teardown": {"duration": 0.0007875759911257774, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManager::test_initialize_providers_missing_config", "lineno": 71, "outcome": "passed", "keywords": ["test_initialize_providers_missing_config", "asyncio", "pytestmark", "TestProviderManager", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006023360037943348, "outcome": "passed"}, "call": {"duration": 0.0009377789974678308, "outcome": "passed"}, "teardown": {"duration": 0.0003557519958121702, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_success", "lineno": 110, "outcome": "passed", "keywords": ["test_complete_with_fallback_success", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005275549920042977, "outcome": "passed"}, "call": {"duration": 0.010758412987343036, "outcome": "passed"}, "teardown": {"duration": 0.0006145590014057234, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_chain", "lineno": 127, "outcome": "failed", "keywords": ["test_complete_with_fallback_chain", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006584319926332682, "outcome": "passed"}, "call": {"duration": 0.0003929509985027835, "outcome": "failed", "crash": {"path": "/home/marku/Documents/programming/LocalProgramming/tests/mocks/mock_provider.py", "lineno": 63, "message": "ValueError: Scenario 'failing' not found"}, "traceback": [{"path": "tests/unit/test_provider_manager.py", "lineno": 132, "message": "in test_complete_with_fallback_chain"}, {"path": "tests/mocks/mock_provider.py", "lineno": 63, "message": "in set_scenario"}], "longrepr": "tests/unit/test_provider_manager.py:132: in test_complete_with_fallback_chain\n    self.mock_ollama.set_scenario(\"failing\")\ntests/mocks/mock_provider.py:63: in set_scenario\n    raise ValueError(f\"Scenario '{name}' not found\")\nE   ValueError: Scenario 'failing' not found"}, "teardown": {"duration": 0.00048474399955011904, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_all_fail", "lineno": 153, "outcome": "passed", "keywords": ["test_complete_with_fallback_all_fail", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005405190022429451, "outcome": "passed"}, "call": {"duration": 0.30226188599772286, "outcome": "passed", "stdout": "Provider ollama failed: name 'aiohttp' is not defined\nProvider openai failed: name 'aiohttp' is not defined\nProvider gemini failed: name 'aiohttp' is not defined\n"}, "teardown": {"duration": 0.0005445169954327866, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_complete_with_fallback_unhealthy_providers", "lineno": 173, "outcome": "passed", "keywords": ["test_complete_with_fallback_unhealthy_providers", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006009839999023825, "outcome": "passed"}, "call": {"duration": 1.0016709229967091, "outcome": "passed"}, "teardown": {"duration": 0.0005540749989449978, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_get_provider", "lineno": 190, "outcome": "passed", "keywords": ["test_get_provider", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00044673199590761214, "outcome": "passed"}, "call": {"duration": 0.00024546298664063215, "outcome": "passed"}, "teardown": {"duration": 0.00016968100680969656, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_get_all_models", "lineno": 198, "outcome": "passed", "keywords": ["test_get_all_models", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006005820032441989, "outcome": "passed"}, "call": {"duration": 0.00033831800101324916, "outcome": "passed"}, "teardown": {"duration": 0.0004198120004730299, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_get_all_models_with_failures", "lineno": 211, "outcome": "passed", "keywords": ["test_get_all_models_with_failures", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0005056240042904392, "outcome": "passed"}, "call": {"duration": 0.0003560610057320446, "outcome": "passed", "stdout": "Error getting models from ollama: Failed to list models\n"}, "teardown": {"duration": 0.00033791699388530105, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerWithMocks::test_health_check_all", "lineno": 227, "outcome": "passed", "keywords": ["test_health_check_all", "asyncio", "pytestmark", "TestProviderManagerWithMocks", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00048588699428364635, "outcome": "passed"}, "call": {"duration": 0.0003040330047952011, "outcome": "passed"}, "teardown": {"duration": 0.0003085320058744401, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic::test_fallback_order_preferred_provider", "lineno": 249, "outcome": "passed", "keywords": ["test_fallback_order_preferred_provider", "asyncio", "pytestmark", "TestProviderManagerFallbackLogic", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.00045699199836235493, "outcome": "passed"}, "call": {"duration": 0.20131381299870554, "outcome": "passed"}, "teardown": {"duration": 0.0006006530020385981, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic::test_fallback_with_mixed_health_states", "lineno": 294, "outcome": "passed", "keywords": ["test_fallback_with_mixed_health_states", "asyncio", "pytestmark", "TestProviderManagerFallbackLogic", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0007586510037072003, "outcome": "passed"}, "call": {"duration": 0.20124384701193776, "outcome": "passed", "stdout": "Provider primary failed: name 'aiohttp' is not defined\n"}, "teardown": {"duration": 0.0006709959998261184, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerFallbackLogic::test_provider_error_handling", "lineno": 323, "outcome": "passed", "keywords": ["test_provider_error_handling", "asyncio", "pytestmark", "TestProviderManagerFallbackLogic", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006324429996311665, "outcome": "passed"}, "call": {"duration": 0.3015051459951792, "outcome": "passed", "stdout": "Provider network_fail failed: name 'aiohttp' is not defined\nProvider auth_fail failed: Authentication failed\n"}, "teardown": {"duration": 0.0005312819994287565, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance::test_concurrent_requests", "lineno": 364, "outcome": "passed", "keywords": ["test_concurrent_requests", "asyncio", "pytestmark", "TestProviderManagerPerformance", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006205410027178004, "outcome": "passed"}, "call": {"duration": 0.011254678000113927, "outcome": "passed"}, "teardown": {"duration": 0.0007069440034683794, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance::test_provider_selection_performance", "lineno": 394, "outcome": "passed", "keywords": ["test_provider_selection_performance", "asyncio", "pytestmark", "TestProviderManagerPerformance", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006788399914512411, "outcome": "passed"}, "call": {"duration": 0.0017577249964233488, "outcome": "passed"}, "teardown": {"duration": 0.0004060860082972795, "outcome": "passed"}}, {"nodeid": "tests/unit/test_provider_manager.py::TestProviderManagerPerformance::test_memory_usage_with_many_providers", "lineno": 425, "outcome": "passed", "keywords": ["test_memory_usage_with_many_providers", "asyncio", "pytestmark", "TestProviderManagerPerformance", "test_provider_manager.py", "unit", "tests", "LocalProgramming", ""], "setup": {"duration": 0.0006042799941496924, "outcome": "passed"}, "call": {"duration": 0.003452913006185554, "outcome": "passed"}, "teardown": {"duration": 0.0008262679912149906, "outcome": "passed"}}], "warnings": [{"message": "Unknown pytest.mark.integration - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html", "category": "PytestUnknownMarkWarning", "when": "collect", "filename": "/home/marku/Documents/programming/LocalProgramming/tests/unit/test_provider_interfaces.py", "lineno": 534}]}